diff --git a/node_modules/discord-player/src/Player.js b/node_modules/discord-player/src/Player.js
index c256c30..07ed96c 100644
--- a/node_modules/discord-player/src/Player.js
+++ b/node_modules/discord-player/src/Player.js
@@ -326,7 +326,7 @@ class Player extends EventEmitter {
             const track = playlist.tracks.shift()
             const queue = await this._createQueue(message, track).catch((e) => this.emit('error', e, message))
             this.emit('trackStart', message, queue.tracks[0])
-            this._addTracksToQueue(message, playlist.tracks)
+            this._addTracksToQueue(message, playlist.tracks, queue)
         }
     }
 
@@ -371,7 +371,7 @@ class Player extends EventEmitter {
                 this.emit('trackAdd', message, queue, queue.tracks[queue.tracks.length - 1])
             } else {
                 const queue = await this._createQueue(message, trackToPlay)
-                this.emit('trackStart', message, queue.tracks[0])
+                this.emit('trackStart', message, queue.tracks[0], queue)
             }
         }
     }
diff --git a/node_modules/discord-player/typings/index.d.ts b/node_modules/discord-player/typings/index.d.ts
index 20dbd45..c1a795b 100644
--- a/node_modules/discord-player/typings/index.d.ts
+++ b/node_modules/discord-player/typings/index.d.ts
@@ -77,7 +77,7 @@ declare module 'discord-player' {
         noResults: [Message, string];
         playlistAdd: [Message, Queue, Playlist];
         trackAdd: [Message, Queue, Track];
-        trackStart: [Message, Track];
+        trackStart: [Message, Track, Queue];
         botDisconnect: [Message];
         channelEmpty: [Message, Queue];
         musicStop: [];
